vms/platformvm/state/staker_diff_iterator.go

```diff
@@ -114,74 +112,57 @@ func (it *stakerDiffIterator) advancePending() {
 type mutableStakerIterator struct {
 	iteratorExhausted bool
 	iterator          StakerIterator
-	heap              []*Staker
+	heap              heap.Queue[*Staker]
+}
+
+func newStakerHeap() heap.Queue[*Staker] {
+	return heap.NewQueue((*Staker).Less)
 }
```

Comment:
Not really sure that this function adds much - feels like we can just inline it in the 2 places we use it